tests/test_backups.py

```diff
@@ -0,0 +1,240 @@
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test backups freeze API."""
+    responses.post(f"{SUPERVISOR_URL}/backups/freeze", status=200)
+    assert await supervisor_client.backups.freeze() is None
+    assert responses.requests.keys() == {
+        ("POST", URL(f"{SUPERVISOR_URL}/backups/freeze"))
+    }
+
+
+async def test_backups_thaw(
+    responses: aioresponses, supervisor_client: SupervisorClient
+) -> None:
+    """Test backups thaw API."""
+    responses.post(f"{SUPERVISOR_URL}/backups/thaw", status=200)
+    assert await supervisor_client.backups.thaw() is None
+    assert responses.requests.keys() == {
+        ("POST", URL(f"{SUPERVISOR_URL}/backups/thaw"))
+    }
+
+
+async def test_partial_backup_options() -> None:
+    """Test partial backup options."""
+    assert PartialBackupOptions(name="good", addons={"a"})
+    assert PartialBackupOptions(name="good", folders={Folder.SSL})
+    assert PartialBackupOptions(name="good", homeassistant=True)
+    with pytest.raises(TypeError):
+        PartialBackupOptions(name="bad")
+
+
+async def test_partial_restore_options() -> None:
+    """Test partial restore options."""
+    assert PartialRestoreOptions(addons={"a"})
+    assert PartialRestoreOptions(folders={Folder.SSL})
+    assert PartialRestoreOptions(homeassistant=True)
+    with pytest.raises(TypeError):
+        PartialRestoreOptions(background=True)
+
+
+def backup_callback(url: str, **kwargs: dict[str, Any]) -> CallbackResult:  # noqa: ARG001
+    """Return response based on whether backup was in background or not."""
+    if kwargs["json"]["background"]:
+        fixture = "backup_background.json"
+    else:
+        fixture = "backup_foreground.json"
+    return CallbackResult(status=200, body=load_fixture(fixture))
+
+
+@pytest.mark.parametrize(
+    ("background", "slug"),
+    [(True, None), (False, "9ecf0028")],
+)
+async def test_backups_full_backup(
+    responses: aioresponses,
+    supervisor_client: SupervisorClient,
+    background: bool,  # noqa: FBT001
+    slug: str | None,
+) -> None:
+    """Test backups full backup API."""
+    responses.post(
+        f"{SUPERVISOR_URL}/backups/new/full",
+        callback=backup_callback,
+    )
+    result = await supervisor_client.backups.full_backup(
+        FullBackupOptions(name="test", background=background)
+    )
+    assert result.job_id == "dc9dbc16f6ad4de592ffa72c807ca2bf"
+    assert result.slug == slug
+
+
+@pytest.mark.parametrize(
+    ("background", "slug"),
+    [(True, None), (False, "9ecf0028")],
+)
+async def test_backups_partial_backup(
+    responses: aioresponses,
+    supervisor_client: SupervisorClient,
+    background: bool,  # noqa: FBT001
+    slug: str | None,
+) -> None:
+    """Test backups full backup API."""
+    responses.post(
+        f"{SUPERVISOR_URL}/backups/new/partial",
+        callback=backup_callback,
+    )
+    result = await supervisor_client.backups.partial_backup(
```

Comment:
Test passing options too?

Reply:
This test passes options already as they are not optional for partial backups.